src/circuit.rs

```diff
@@ -41,6 +42,69 @@ impl From<JubJubAffine> for PublicInputValue {
     }
 }
 
+#[derive(Debug, Clone)]
+/// Collection of structs/objects that the Verifier will use in order to
+/// de/serialize data needed for Circuit proof verification.
+/// This structure can be seen as a link between the [`Circuit`] public input
+/// positions and the [`VerifierKey`] that the Verifier needs to use.
+pub struct VerifierData {
+    pub(self) key: VerifierKey,
```

Comment:
`pub(self)` shouldn't be required. It's equivalent to omit the visibility at all.